### PR TITLE
Add ability to process multi-document YAML files

### PR DESCRIPTION
**Pull request by ssbarnea**
Closes: #2288

### CODE REVIEW

File: src/ansiblelint/utils.py [hunk 1]
```diff
@@ -709,12 +709,15 @@ def get_action_tasks(data: AnsibleBaseYAMLObject, file: Lintable) -> List[Any]:
     ]
 
 
-@lru_cache(maxsize=128)
-def parse_yaml_linenumbers(lintable: Lintable) -> AnsibleBaseYAMLObject:
+@lru_cache(maxsize=0)
```
Comment (cognifloyd) [inline src/ansiblelint/utils.py]: Does 0 make the cache unlimited here?

**Review by cognifloyd — COMMENTED** [review]
Cool! I hadn't considered multi document yaml files. I wonder if `--write` messes them up.

File: src/ansiblelint/utils.py (hunk 1 above)
Comment (ssbarnea) [reply src/ansiblelint/utils.py]: Fixed it to None to make it unlimited after reading the docs. To my surprise they used None instead of 0.

**Comment by ssbarnea** [comment]
> Cool! I hadn't considered multi document yaml files. I wonder if `--write` messes them up.

Multi-documents are extremely uncommon, but you might want to add some code to avoid messing them if you encounter them. I spotted on inside debops repo, that is why I made this fix.

**Comment by cognifloyd** [comment]
Another example: I frequently see multi-document yaml with kubernetes resources.

**Pull request merged**